keylime/ima/ima.py

```diff
@@ -322,7 +321,7 @@ def _process_measurement_list(
     errors: Dict[Type[ast.Mode], int] = {}
     pcrval_bytes = b""
     if pcrval is not None:
-        pcrval_bytes = codecs.decode(pcrval.encode("utf-8"), "hex")
+        pcrval_bytes = bytes.fromhex(pcrval)
```

Comment:
My guess here is we didn't know about the fromhex() method on bytes and therefore used the more complicated code...

Reply:
```
>>> bytes.fromhex("123456abcd")
b'\x124V\xab\xcd'
>>> codecs.decode("123456abcd".encode("utf-8"), "hex")
b'\x124V\xab\xcd'
```

Even number of letters required:
```
>>> bytes.fromhex("123456abc")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: non-hexadecimal number found in fromhex() arg at position 9

>>> codecs.decode("123456abc".encode("utf-8"), "hex")
Traceback (most recent call last):
  File "/usr/lib64/python3.10/encodings/hex_codec.py", line 19, in hex_decode
    return (binascii.a2b_hex(input), len(input))
binascii.Error: Odd-length string

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
binascii.Error: decoding with 'hex' codec failed (Error: Odd-length string)
```

non-hex letters make both fail:
```
>>> bytes.fromhex("123456gg")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: non-hexadecimal number found in fromhex() arg at position 6

>>> codecs.decode("123456gg".encode("utf-8"), "hex")
Traceback (most recent call last):
  File "/usr/lib64/python3.10/encodings/hex_codec.py", line 19, in hex_decode
    return (binascii.a2b_hex(input), len(input))
binascii.Error: Non-hexadecimal digit found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
binascii.Error: decoding with 'hex' codec failed (Error: Non-hexadecimal digit found)
```

Reply:
Note: We should probably do this throughout the whole code base where codec is used for writing hex numbers or parsing hex numbers

Reply:
Yes that was the case, we just need to make sure that the behavior is the same and now catch a  `ValueError` instead of a `binascii.Error`.